rasa/utils/tensorflow/rasa_layers.py

```diff
@@ -28,9 +29,182 @@
+        new_sparse_feature_sizes: List[int], old_sparse_feature_sizes: List[int],
+    ) -> None:
+        """Checks if the sizes of sparse features have decreased during fine-tuning.
+
+        Sparse feature sizes might decrease after changing the training data.
+        This can happen for example with `LexicalSyntacticFeaturizer`.
+        We don't support this behaviour and we raise an exception if this happens.
+
+        Args:
+            new_sparse_feature_sizes: sizes of current sparse features for a
+                                      specific attribute and feature type.
+            old_sparse_feature_sizes: sizes of sparse features the model was trained on
+                                      before for a specific attribute and feature type.
+
+        Raises:
+            RasaException: When any of the sparse feature sizes decrease
+                           from the last time training was run.
+        """
+        for new_size, old_size in zip(
+            new_sparse_feature_sizes, old_sparse_feature_sizes
+        ):
+            if new_size < old_size:
+                raise RasaException(
+                    "Sparse feature sizes have decreased from the last time training "
+                    "was run. The training data was changed in a way that resulted in "
+                    "some features not being present in the data anymore. This can "
+                    "happen if you had `LexicalSyntacticFeaturizer` in your pipeline. "
+                    "The pipeline cannot support incremental training in this setting."
+                    " We recommend you to retrain the model from scratch."
+                )
+
+    @staticmethod
+    def _replace_dense_for_sparse_layer(
+        layer_to_replace: layers.DenseForSparse,
+        new_sparse_feature_sizes: List[int],
+        old_sparse_feature_sizes: List[int],
+        attribute: Text,
+        feature_type: Text,
+        reg_lambda: float,
+    ) -> layers.DenseForSparse:
+        """Replaces a `DenseForSparse` layer with a new one.
+
+        Replaces an existing `DenseForSparse` layer with a new one
+        in order to adapt it to incremental training.
+
+        Args:
+            layer_to_replace: a `DenseForSparse` layer that is used to create a new one.
+            new_sparse_feature_sizes: sizes of sparse features that will be
+                                      the input of the layer.
+            old_sparse_feature_sizes: sizes of sparse features that used to be
+                                      the input of the layer.
+            attribute: an attribute of the data fed to the layer.
+            feature_type: a feature type of the data fed to the layer.
+            reg_lambda: regularization constant.
+
+        Returns:
+            New `DenseForSparse` layer.
+        """
+        kernel = layer_to_replace.get_kernel().numpy()
+        bias = layer_to_replace.get_bias()
+        use_bias = False if bias is None else True
```

Comment:
In python if you want to check if a variable `x` is empty / None, you don't need to equate it to something. you can just write `if x` or `if not x`.
```suggestion
        use_bias = False if not bias else True
```

Reply:
That's how I wrote it the first time but as it's a tf.tensor it gave me this error:
`ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()`